common/utils.js

```diff
@@ -60,16 +60,86 @@
             // Then, parse the URL fragment id (aka, hash). Expected format:
             //  "#catalog_id/[schema_name:]table_name[/{attribute::op::value}{&attribute::op::value}*][@sort(column[::desc::])]"
             var hash = location.hash;
+            var uri = hash;
             if (hash === undefined || hash == '' || hash.length == 1) {
                 return;
             }
 
-            // parse out @sort(...)
+            // parse out modifiers
             if (hash.indexOf("@sort(") !== -1) {
-                context.sort = hash.match(/@sort\((.*)\)/)[1];
-                hash = hash.split("@sort(")[0];
+                hash = hash.split("@sort(")[0]; // remove modifiers from uri
+            } else if (hash.indexOf("@before(") !== -1) {
+                hash = hash.split("@before(")[0]; // remove modifiers from uri
+            } else if (hash.indexOf("@after(") !== -1) {
+                hash = hash.split("@after(")[0]; // remove modifiers from uri
+            } else if (hash.indexOf("?limit=") !== -1) {
+                hash = hash.split("?limit=")[0]; // remove modifiers from uri
             }
```

Comment:
The repetition of `if hash.indexOf(some_modifier) !== -1 ... hash = hash.split(some_modifier)[0]` seems like a good candidate for a currying function here, especially if there will be additional modifiers.